opentelemetry-api/src/opentelemetry/util/_decorator.py

```diff
@@ -78,4 +78,12 @@ def _agnosticcontextmanager(
     def helper(*args: Pargs, **kwargs: Pkwargs) -> _AgnosticContextManager[R]:
         return _AgnosticContextManager(func, args, kwargs)
 
+    # Ingoring the type to keep the original signature of the function
+    return helper  # type: ignore[return-value]
+    func: "Callable[P, Iterator[R]]",
+) -> "Callable[P, _AgnosticContextManager[R]]":  # type: ignore[return-value]
+    @functools.wraps(func)
+    def helper(*args: Pargs, **kwargs: Pkwargs) -> _AgnosticContextManager[R]:
+        return _AgnosticContextManager(func, args, kwargs)
+
     return helper
```

Comment:
Sorry @xrmx my "suggestion" was not multiline :cry: 

Here is the right code you should put instead of the current `_agnosticcontextmanager`.

```python
def _agnosticcontextmanager(
    func: "Callable[P, Iterator[R]]",
) -> "Callable[P, _AgnosticContextManager[R]]":
    @functools.wraps(func)
    def helper(*args: Pargs, **kwargs: Pkwargs) -> _AgnosticContextManager[R]:
        return _AgnosticContextManager(func, args, kwargs)

    # Ingoring the type to keep the original signature of the function
    return helper  # type: ignore[return-value]
```

Reply:
Thanks!